guides/source/generators.md

```diff
@@ -475,35 +496,255 @@ after_bundle do
+
+Adds a line inside the `Application` class for `config/application.rb`.
+
+If `options[:env]` is specified, the line is appended to the corresponding file in `config/environments`.
+
+```ruby
+environment 'config.action_mailer.default_url_options = {host: "http://yourwebsite.example.com"}', env: "production"
+```
+
+The above will add the config line to `config/environments/production.rb`.
+
+### initializer, vendor, lib, file
+
+The `initializer` helper method adds an initializer to the generated application's `config/initializers` directory.
+
+After adding the below to the `template.rb` file, you can use `Object#not_nil?` and `Object#not_blank?` in your application:
+
+```ruby
+initializer "not_methods.rb", <<-CODE
+  class Object
+    def not_nil?
+      !nil?
+    end
+
+    def not_blank?
+      !blank?
+    end
+  end
+CODE
+```
+
+Similarly, the `lib()` method creates a file in the `lib/` directory and `vendor()` method creates a file in the `vendor/` directory.
+
+There is also a `file()` method, which accepts a relative path from `Rails.root` and creates all the directories and files needed:
+
+```ruby
+file "app/components/foo.rb", <<-CODE
+  class Foo
+  end
+CODE
+```
+
+The above will create the `app/components` directory and put `foo.rb` in there.
+
+### rakefile
+
+The `rakefile` method creates a new rake file under `lib/tasks` with the given tasks:
+
+```ruby
+rakefile("bootstrap.rake") do
+  <<-TASK
+    namespace :boot do
+      task :strap do
+        puts "i like boots!"
+      end
+    end
+  TASK
+end
+```
+
+The above creates `lib/tasks/bootstrap.rake` with a `boot:strap` rake task.
+
+### generate
+
+You can even call a generator from inside a `template.rb`. The following runs the `scaffold` rails generator with the given arguments:
+
+```ruby
+generate(:scaffold, "person", "name:string", "address:text", "age:number")
+```
+
+### run
+
+This method executes an arbitrary command. Let's say you want to remove the `README.rdoc` file:
+
+```ruby
+run "rm README.rdoc"
+```
+
+### rails_command
+
+You can run the Rails commands in the generated application. Let's say you want to migrate the database at some point in the template ruby script:
+
+```ruby
+rails_command "db:migrate"
+```
+
+You can also run commands with a different Rails environment:
+
+```ruby
+rails_command "db:migrate", env: "production"
+```
+
+You can also run commands that should abort application generation if they fail:
+
+```ruby
+rails_command "db:migrate", abort_on_failure: true
+```
+
+### route
+
+This method adds an entry to the `config/routes.rb` file. To make `PeopleController#index` the default page for the application, we can add:
+
+```ruby
+route "root to: 'person#index'"
+```
+
+### inside
+
+This mehtod enables you to run a command from a given directory. For example, if you have a copy of edge rails that you wish to symlink from your new apps, you can do this:
+
+```ruby
+inside("vendor") do
+  run "ln -s ~/commit-rails/rails rails"
```

Comment:
Nitpick, but maybe the path to the local version of edge rails could be a bit more clearer?
For example:
```suggestion
  run "ln -s ~/my-forks/rails rails"
```